ballista/rust/executor/src/main.rs

```diff
@@ -148,3 +167,108 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+/// This function will scheduled periodically for cleanup executor.
+/// Will only clean the dir under work_dir not include file
+async fn clean_shuffle_data_loop(work_dir: &str, seconds: i64) -> Result<()> {
+    let mut dir = fs::read_dir(work_dir).await?;
+    let mut to_deleted = Vec::new();
+    let mut need_delete_dir;
+    while let Some(child) = dir.next_entry().await.unwrap() {
+        if let Ok(metadata) = child.metadata().await {
+            if metadata.is_dir() {
+                let dir = fs::read_dir(child.path()).await?;
+                match check_modified_time_in_dirs(vec![dir], seconds).await {
+                    Ok(x) => match x {
+                        true => {
+                            need_delete_dir = child.path().into_os_string();
+                            to_deleted.push(need_delete_dir)
+                        }
+                        false => {}
+                    },
+                    Err(e) => {
+                        error!("Fail in clean_shuffle_data_loop {:?}", e)
+                    }
+                }
+            }
+        } else {
+            error!("can not get meta from file{:?}", child)
+        }
+    }
+    info!(
+        "Executor work_dir {:?} not modified in {:?} seconds will be deleted ",
+        &to_deleted, seconds
+    );
+    for del in to_deleted {
+        fs::remove_dir_all(del).await?;
+    }
+    Ok(())
+}
+
+/// Determines if a directory all files are older than cutoff seconds.
+async fn check_modified_time_in_dirs(
```

Comment:
Is it possible that during a long running job all the files are over the threshold but the job is still not finished?

Reply:
If it is possible, maybe we can also add a flag in the job folder to indicate job status.

Reply:
> Is it possible that during a long running job all the files are over the threshold but the job is still not finished?
I thinks it will solved by the set proper TTL args.

> If it is possible, maybe we can also add a flag in the job folder to indicate job status.
IMHO, executor never know a job is finished, only scheduler has a loop to check all tasks in one job. 
If add a indicate, we need add info exchange from scheduler to executor which  i am not prefer.

Reply:
Spark default set TTL 7 days,  `This is a Time To Live and should depend on the amount of available disk space you have.`  i think if we have a small disk, long time job will be deleted and forbidden for avoid affect other job.

Reply:
I also prefer to keep the communication between scheduler and executor simple. Spark writes a file in the folder after stage succeeded. This handles some error cases but not all. It may worth to do something similar,  in the case you mentioned, if we have a small disk and need to clear some space to run other tasks, we can delete finished stages and clean up the test with TTL. Either way I don't feel we need to block this PR, we may create an issue to capture as future improvement.

Reply:
Thanks 🤝  nice talk with you!